Rectangle.cpp

```diff
@@ -1,15 +1,48 @@
 #ifndef Rectangle_cpp
 #define Rectangle_cpp
+
 #include <iostream>
-#include "Draw.cpp"
+#include "Shape.cpp"
+#include "Area.cpp"
+#include "Input.cpp"
+
 using namespace std;
 
-class Rectangle : public Draw
+class Rectangle : public Shape, public Input
 {
-public:
-    void drawShape() override
-    {
-        cout << "rectangle" << endl;
-    }
+    private:
+        double length = input();
```

Comment:
Again, we are coupling the creation of the object with the user input for the fields.